pandas/tests/indexes/datetimes/test_partial_slicing.py

```diff
@@ -330,3 +330,9 @@ def test_loc_datetime_length_one(self):
 
         result = df.loc['2016-10-01T00:00:00':]
         tm.assert_frame_equal(result, df)
+
+    def test_comparison_to_datetime64(self):
+        # GH issue #17965
+        df = DataFrame({'dates': date_range('2000-01-01', '2000-01-05')})
+        date = df['dates'].unique()[0]
```

Comment:
Is this the right test? `date` should be a `datetime.date`, and I think this is a `datetime64[ns]`.

Reply:
date is `numpy.datetime64('2000-01-01T00:00:00.000000000')`, which is consistent with the output of `pd.date_range().values`, which are `datetime64[ns]` too.  Do you suggest this should be an array of `datetime.date` objects?

Reply:
#17965 is about comparing with a single `datetime.date` object:

```python
In [20]: pd.Timestamp('20130101').to_pydatetime().date()>=df.A
ValueError: cannot set a Timestamp with a non-timestamp
```

Reply:
paramaterize this test with ``date, datetime, np.datetime64, and Timestamp``.
check for < and ==

Reply:
True! I was not actually fixing the Issue, but rather this comment: https://github.com/pandas-dev/pandas/issues/17965#issuecomment-342821038 - but it actually applies to both.